### PR TITLE
[stable/jenkins] add backup cronjob

### LINKED ISSUE #9987: [stable/jenkins] Backup/restore feature
**Is this a request for help?**:
no

**Is this a BUG REPORT or FEATURE REQUEST?**:
Feature request

Adding a backup/restore mecanism could be great in the Jenkins chart (I had plenty of issues with backup jenkins plugins). Actually I implemented it quick and dirty with AWS S3 as a storage but never opensourced it. Is the community interested in this? I saw a rotten feature request #3344. 

The configuration I created in the values.yaml looked like this:

```
backup:
  aws_access_key_id: "${BACKUP_ACCESS_KEY_ID}"
  aws_access_key_secret: "${BACKUP_ACCESS_KEY_SECRET}"
  retrieve:
    enabled: false
    name: jenkins.tar.gz
    path: "s3://${BACKUP_S3_BUCKET}/jenkins"
    image:
      repository: "devopsexperience/backup-retriever"
      tag: "1.0"
  upload:
    enabled: true
    name: jenkins.tar.gz
    path: "s3://${BACKUP_S3_BUCKET}/jenkins"
    schedule: "0 10 * * *"
    image:
      repository: "devopsexperience/backup-companion"
      tag: "1.0"
```

What the chart does is spawn a sidecar container which backups on schedule "/var/jenkins_home". And for the restore part, it is an initcontainer which fetches the archive, untar it then let the copy-default-config do its job.

I think, this gives the user an easy way to override too (change the container image). Will need to find a better way of passing the cloud provider credentials though.

If you are interested, I am willing to submit a PR.

### PR DESCRIPTION
**Pull request by maorfr**
Signed-off-by: Maor <maor.friedman@nuvo-group.com>

#### What this PR does / why we need it:
Adds a backup CronJob for Jenkins

#### Which issue this PR fixes
  - fixes #9987

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by maorfr** [comment]
/assign @viglesiasce 
Thanks!

File: stable/jenkins/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: jenkins
 home: https://jenkins.io/
-version: 0.26.2
+version: 0.26.3
```
Comment (viglesiasce) [inline stable/jenkins/Chart.yaml]: Lets do a bump to 0.27.0

File: stable/jenkins/README.md [hunk 2]
```diff
@@ -236,6 +236,31 @@ If running upon a cluster with RBAC enabled you will need to do the following:
...
+Adds a backup CronJob for jenkins, along with required RBAC resources.
+
+### Backup Values
+
+| Parameter                   | Description                                | Default                           |
+| --------------------------- | ------------------------------------------ | --------------------------------- |
+| `backup.enabled`            | Enable the use of a backup CronJob         | `false`                           |
+| `backup.schedule`           | Schedule to run jobs                       | `0 2 * * *`                       |
+| `backup.annotations`        | Backup pod annotations                     | iam.amazonaws.com/role: `jenkins` |
+| `backup.image.repo`         | Backup image repository                    | `nuvo/kube-tasks`                 |
```
Comment (viglesiasce) [inline stable/jenkins/README.md:250]: This image needs to be added to sources.

**Review by viglesiasce — CHANGES_REQUESTED** [review]

File: stable/jenkins/Chart.yaml (hunk 1 above)
Comment (maorfr) [reply stable/jenkins/Chart.yaml]: done

File: stable/jenkins/README.md (hunk 2 above)
Comment (maorfr) [reply stable/jenkins/README.md:250]: done

**Review by viglesiasce — APPROVED** [review]

**Pull request merged**